src/reactapp/src/components/shippingMethod/components/ShippingMethodFormManager.jsx

```diff
@@ -1,34 +1,40 @@
-import React, { useEffect } from 'react';
-import { node } from 'prop-types';
+import React, { useEffect, useMemo, useState } from 'react';
 import { Form } from 'formik';
+import { node } from 'prop-types';
 import { string as YupString } from 'yup';
 
 import { __ } from '../../../i18n';
+import { _isObjEmpty } from '../../../utils';
 import { SHIPPING_METHOD } from '../../../config';
 import useFormSection from '../../../hook/useFormSection';
 import { formikDataShape } from '../../../utils/propTypes';
+import useCheckoutFormContext from '../../../hook/useCheckoutFormContext';
 import ShippingMethodFormContext from '../context/ShippingMethodFormContext';
 import useShippingMethodAppContext from '../hooks/useShippingMethodAppContext';
 import useShippingMethodCartContext from '../hooks/useShippingMethodCartContext';
-import { _isObjEmpty } from '../../../utils';
 
-const initialValues = {
+const defaultValues = {
   methodCode: '',
   carrierCode: '',
 };
 
 const requiredMessage = __('Required');
 
-const validationSchema = {
+const initialValidationSchema = {
   methodCode: YupString().required(requiredMessage),
   carrierCode: YupString().required(requiredMessage),
 };
 
 function ShippingMethodFormManager({ children, formikData }) {
+  const { setFieldValue } = formikData;
+  const [initialValues, setInitialValues] = useState(defaultValues);
+  const [validationSchema, setValidationSchema] = useState(
```

Comment:
Turn `validationSchema` into a state will be beneficial especially when the shipping integrations needs to update the validation schema of shipping section altogether.